textuilib2.py

```diff
@@ -89,18 +87,14 @@ def clear():
     os.system(clr)
 
 def choice(menu):
-    print ''
-    print 'Choose an Option:'
+    print('')
+    print('Choose an Option:')
     try:
-        option = input('Option: ')
-        i = 0
-        for o in menu.menuItems:
-            if i == option - 1:
-                return o.function
-            i += 1
+        option = int(input('Option: '))
+        if option-1 < len(menu.menuItems):
+            return menu.menuItems[option-1].function
+        else:
+            raise ValueError("Invalid option!")
 
     except (KeyboardInterrupt, SystemExit):
         raise
-
-    except:
-        return
```

Comment:
I don't understand why you removed these three lines. They were there for looping after an incorrect option.